### PR TITLE
Fix __eq__ and __repr__ when using resolver fields

### PR DESCRIPTION
**Pull request by patrick91**
## Description

Eq and repr where broken previously. We don't want to include fields with resolvers for eq and repr
as they are basically functions and doesn't make sense to compare functions :)

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This relases fixes an issue with the generated `__eq__` and `__repr__` methods when defining
fields with resolvers.

This now works properly:

```python
@strawberry.type
class Query:
    a: int

    @strawberry.field
    def name(self) -> str:
        return "A"

assert Query(1) == Query(1)
assert Query(1) != Query(2)
```

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This relases fixes an issue with the generated `__eq__` and `__repr__` methods when defining
fields with resolvers.

This now works properly:

```python
@strawberry.type
class Query:
    a: int

    @strawberry.field
    def name(self) -> str:
        return "A"

assert Query(1) == Query(1)
assert Query(1) != Query(2)
```

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This relases fixes an issue with the generated `__eq__` and `__repr__` methods when defining
fields with resolvers.

This now works properly:

```python
@strawberry.type
class Query:
    a: int

    @strawberry.field
    def name(self) -> str:
        return "A"

assert Query(1) == Query(1)
assert Query(1) != Query(2)
```

**Review by jkimbo — APPROVED** [review]

File: strawberry/field.py [hunk 1]
```diff
@@ -91,6 +94,9 @@ def __call__(self, resolver: _RESOLVER_TYPE) -> "StrawberryField":
         self.base_resolver = resolver
         self.type = resolver.type
 
+        self.repr = False
+        self.compare = False
+
```
Comment (BryceBeagle) [inline strawberry/field.py:99]: This is done in `@base_resolver.setter`. Is that not working?

Comment (BryceBeagle) [reply strawberry/field.py:99]: Here: https://github.com/strawberry-graphql/strawberry/blob/f14305b6b2918424f2ac70519a61617c7136f113/strawberry/field.py#L130-L132

Comment (patrick91) [reply strawberry/field.py:99]: uh, I missed it, but no it is not working, let me check why

Comment (patrick91) [reply strawberry/field.py]: oh we were missing `compare`, updated!

File: strawberry/field.py [hunk 2]
```diff
@@ -127,8 +130,9 @@ def base_resolver(self, resolver: StrawberryResolver) -> None:
         self._base_resolver = resolver
         self.origin = resolver.wrapped_func
 
-        # Don't add field to __init__ or __repr__ once it has a resolver
+        # Don't add field to __init__, __repr__ and __eq__ once it has a resolver
         self.init = False
+        self.compare = False
```
Comment (BryceBeagle) [inline strawberry/field.py:135]: Should we not be comparing fields with resolvers? If I have a type with only fields with resolvers, should instances _always_ be equal?

Comment (patrick91) [reply strawberry/field.py:135]: I'd say yes, how would you compare resolvers?

Comment (BryceBeagle) [reply strawberry/field.py:135]: Is executing them unreasonable? I guess only if they have no arguments

Comment (patrick91) [reply strawberry/field.py:135]: @BryceBeagle I'll merge this for now, I think it makes sense to compare just values :)

Comment (patrick91) [reply strawberry/field.py:135]: > Is executing them unreasonable? I guess only if they have no arguments

oh sorry, I missed your comment. I'd say so, also there might be side effects too :)

**Pull request merged**